src/wpt/harness/harness.ts

```diff
@@ -714,38 +715,122 @@ async function validate(
   await Promise.all(globalThis.promises);
 
   const expectedFailures = new Set(options.expectedFailures ?? []);
+  const unexpectedFailures = [];
 
-  let failing = false;
   for (const err of globalThis.errors) {
     if (!expectedFailures.delete(err.message)) {
       err.message = sanitize_unpaired_surrogates(err.message);
       console.error(err);
-      failing = true;
+      unexpectedFailures.push(err.message);
     } else if (options.verbose) {
       err.message = sanitize_unpaired_surrogates(err.message);
       console.warn('Expected failure: ', err);
     }
   }
 
-  if (failing) {
-    throw new Error(`${testFileName} failed`);
+  if (unexpectedFailures.length > 0) {
+    console.info(
+      'The following tests unexpectedly failed:',
+      unexpectedFailures
+    );
   }
 
   if (expectedFailures.size > 0) {
-    console.info('Expected failures', expectedFailures);
-    throw new Error(
-      'Expected failures but test succeeded. Please update the test config file.'
+    console.info(
+      'The following tests were expected to fail but instead succeeded:',
+      [...expectedFailures]
     );
   }
+
+  if (unexpectedFailures.length > 0 || expectedFailures.size > 0) {
+    throw new Error(`${testFileName} failed. Please update the test config.`);
+  }
+}
+
+class WPTMetadata {
+  // Specifies the Javascript global scopes for the test. (Not currently supported)
+  public global: string[] = [];
+  // Specifies additional JS scripts to execute.
+  public scripts: string[] = [];
+  // Adjusts the timeout for the tests in this file. (Not currently supported)
+  public timeout?: string;
+  // Sets a human-readable title for the entire test file. (Not currently supported.)
+  public title?: string;
+  // Specifies a variant of this test, which can be used in subsetTestByKey (Not currently supported.)
+  public variants: URLSearchParams[] = [];
+}
+
+function parseWptMetadata(code: string): WPTMetadata {
+  const meta = new WPTMetadata();
+
+  for (const { groups } of code.matchAll(
+    /\/\/ META: (?<key>\w+)=(?<value>.+)/g
+  )) {
+    if (!groups || !groups.key || !groups.value) {
+      continue;
+    }
+
+    switch (groups.key) {
+      case 'global':
+        meta.global = groups.value.split(',');
+        break;
+
+      case 'script': {
+        // Replace leading "./" if found
```

Comment:
What happens if it starts with ../?

Reply:
I switched to `path.normalize` which is a bit nicer, but we still have limitations on what files can be accessed in a wd-test. We'll at least get an error message if a file can't be loaded.
https://github.com/cloudflare/workerd/blob/2cab4bb9dd84ab2b3d0ab6142e6669177aca8748/src/wpt/harness/harness.ts#L829